### ISSUE TITLE
Itemized columns in function not working 

### ISSUE THREAD

**Issue by jwc-clinnection** [sqlc-dev/sqlc#2386]
### Version

1.18.0

### What happened?

When not using * (A_Star) and itemizing columns, functions fail to generate.

### Relevant log output

```shell
sqlc generate failed.

# package db
query.sql:15:8: column "id" does not exist
```

### Database schema

```sql
create table blog (
    id serial primary key,
    name text not null

);

create function test_select_blog(in p_id int)
    returns table (id int, name text) AS $$
BEGIN RETURN QUERY
    select id, name from blog where id = p_id;
END;
$$ language plpgsql;
```

### SQL queries

```sql
-- name: TestFuncSelectBlog :many
select id, name from test_select_blog($1);
```

### Playground URL

https://play.sqlc.dev/p/122897d98d937d01051777c6ab253ccb16e5c20d513c03c4e1049bf36349f4f5

### What operating system are you using?

Linux

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by saquibmian** [comment]
Is there any workaround for this? The best I can think of is to create a fake table that matches the composite table declaration and use `setof <table name>` instead, but this is far from ideal.

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/func_return_table_columns/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/d9f726be3fe164234a418249f53a9875373b23155cfef5edba013b8afa410083)
